### DISCUSSION TITLE
There has the possibility of NPE

### DISCUSSION THREAD

**Discussion by Oliverwqcwrw** [apache/skywalking#8966] [General]
In `skywalking-java` `org.apache.skywalking.apm.agent.core.boot.AgentPackagePath#findPath`

> if (isInJar) {
                urlString = urlString.substring(urlString.indexOf("file:"), insidePathIndex);
                File agentJarFile = null;
                try {
                    agentJarFile = new File(new URL(urlString).toURI());
                } catch (MalformedURLException | URISyntaxException e) {
                    LOGGER.error(e, "Can not locate agent jar file by url:" + urlString);
                }
                if (agentJarFile.exists()) {
                    return agentJarFile.getParentFile();
                }
            }

If `agentJarFile = new File(new URL(urlString).toURI());` occur exception then the agentJarFile is null and `agentJarFile.exists()` can occur NPE

I'm not sure I'm right

**Comment by wu-sheng** [comment]
When and how could make `agentJarFile is null`?

**Comment by Oliverwqcwrw** [comment]
Such as specifying a wrong path or wrong file

**Reply by wu-sheng** [reply]
The wrong path means no file would be found. We never faced this NPE.

**Reply by Oliverwqcwrw** [reply]
Ok, Thanks for your solution
